liquibase-standard/src/main/java/liquibase/diff/output/changelog/core/MissingDataChangeGenerator.java

```diff
@@ -60,15 +62,22 @@ public Change[] fixMissing(DatabaseObject missingObject, DiffOutputControl outpu
                 return null;
             }
 
-            String sql = "SELECT * FROM " + referenceDatabase.escapeTableName(table.getSchema().getCatalogName(), table.getSchema().getName(), table.getName());
```

Comment:
We can check here for the include/exclude params, and if they are not set we can go on with the old code saving 1 query and keeping the old flow. What do you think?

Reply:
Sure, I can include some of the previous logic.

Reply:
Brought back some of the old logic.
Added more test cases.

What shall happen when the exclude/include filters produce an empty column set ?
In my opinion, they shall have no effect, meaning a warning is issued and ALL columns are considered.

Reply:
Maybe it's better to throw an error? This way the user won't miss it.
Also, I think we are not handling case sensitive/insensitive right? I believe we'll need to use `referenceDatabase.isCaseSensitive()` when matching the column names to ensure that we don't match incorrect column names.

Reply:
Good points.
Will throw an exception for no columns identified.
Will think of case sensitivity.

Thanks.

Reply:
One more question, since I use Postresql, is there a way to instantiate a DatabaseTestSystem that is case-sensitive ?

Reply:
Also, aside from Java wild-cards, shall we also support SQL wild-cards ?

Java:  '.', '?', '+', '*', '[', ']' etc.

SQL: '_', '%', ....

We only support Java currently.

Reply:
Added a CommandExecutionException when the exclude/include filters come up short.

Enhanced logic to support case-sensitive columns; it is tricky.
Would like to have a test with a more realistic test DB, one which is case-sensitive.

Reply:
For the test, I think you can overload the MockDb as in https://github.com/liquibase/liquibase/blob/d9376c4110e4f1ac7c5d047fb686b22242c8b78f/liquibase-standard/src/test/groovy/liquibase/change/core/LoadDataChangeTest.groovy#L634 , something like:

```
        def mockDB = new MockDatabase() {
            @Override
            boolean isCaseSensitive() {
                return true
            }
        }
```

Reply:
If you need anything else for the tests, please let me know.

Reply:
`MockDatabase` and `DatabaseTestSystem` are not related, for starters.

It would be great if there was a `MockDatabaseTestSystem`.

Are you not suggesting I introduce a case-sensitive DB in form of a `MockDatabase` to show that the changes also apply to it ?

I do not see how I can use a `MockDatabase`, sorry.
It does not have an `.executeSql` method ....

Reply:
Ok, so if mocking is not an option we can leave it as is, I'm fine with that!